var/spack/repos/builtin/packages/python/package.py

```diff
@@ -103,6 +114,31 @@ def setup_dependent_environment(self, module, spec, ext_spec):
         else:
             module.python = Executable(join_path(spec.prefix.bin, 'python'))
 
+        # The code below patches the any python extension to have good defaults for `setup_dependent_environment` and
+        # `setup_environment` only if the extension didn't override any of these functions explicitly.
+        def _setup_env(self, env):
+            site_packages = glob.glob(join_path(self.spec.prefix.lib, "python*/site-packages"))
```

Comment:
@tgamblin I left the logic here exactly the same it was in develop. Please tell me if you want to change it with something more appropriate.